openrouteservice/src/main/java/org/heigit/ors/routing/RoutingProfileManager.java

```diff
@@ -409,26 +411,49 @@ public RouteResult matchTrack(MapMatchingRequest req) throws Exception {
                 }
             }
 
-            try {
-                for (Object obj : gr.getReturnObjects()) {
-                    if (obj instanceof ExtraInfoProcessor) {
-                        if (extraInfoProcessor == null) {
-                            extraInfoProcessor = (ExtraInfoProcessor)obj;
+            List<ExtraInfoProcessor> extraList = new ArrayList<>();
+            for (Object o : gr.getReturnObjects()) {
```

Comment:
It would be clearer to use a more informative variable name here (and for the others later) such as `returnObject`